src/modules/interfaces.ts

```diff
@@ -782,6 +797,62 @@ export class ModuleBase {
     return [];
   }
 
+  getFunctionsFromSql(sql: string): {
+    [key: string]: { description?: string; sample_usage?: string; signature?: string };
+  } {
+    const returnValue: {
+      [key: string]: { description?: string; sample_usage?: string; signature?: string };
+    } = {};
+    try {
+      const sqlParsed = parse(sql);
+      const functionsAndParams = sqlParsed
+        .filter((s: any) => _.has(s, ['RawStmt', 'stmt', 'CreateFunctionStmt']))
```

Comment:
Nit. `_.has` is most likely slower than:

```suggestion
        .filter((s: any) => !!s?.RawStmt?.stmt?.CreateFunctionStmt))
```

and while it's full of symbols requires a smaller body of knowledge for the reader (because it's just standard modern Javascript) so I think it'll take less time to understand what it's doing.

Reply:
+1